app/ts/library/utils.ts

```diff
@@ -7,3 +7,19 @@ export function addressString(address: bigint): string {
 }
 
 export const min = (left: bigint, right: bigint) => left < right ? left : right
+
+function bigintToUint8Array(value: bigint, numberOfBytes: number) {
+	if (typeof value === 'number') value = BigInt(value)
+	if (value >= 2n ** BigInt(numberOfBytes * 8) || value < 0n) throw new Error(`Cannot fit ${ value } into a ${ numberOfBytes }-byte unsigned integer.`)
+	const result = new Uint8Array(numberOfBytes)
+	for (let i = 0; i < result.length; ++i) {
+		result[i] = Number((value >> BigInt(numberOfBytes - i - 1) * 8n) & 0xffn)
+	}
+	return result
+}
+
+export function stringToUint8Array(data: string) {
```

Comment:
```suggestion
export function hexToUint8Array(data: string) {
```
Recommend renaming to be clear this doesn't work for any strings, it is specifically for hex encoded strings with `0x` prefix.  I considered `prefixedHexToUint8Array` and would accept that as well, but it is notably longer and arguably provides minimal benefit.

If you want to refactor *all* of your code you could change it to:
```suggestion
export function hexToUint8Array(data: `0x{string}) {
```